componenets/org.wso2.carbon.identity.application.authenticator.naver/src/main/java/org.wso2.carbon.identity.application.authenticator.naver/NaverCustomAuthenticator.java

```diff
@@ -46,6 +85,65 @@ public class NaverCustomAuthenticator extends Oauth2GenericAuthenticator {
 
     private static final long serialVersionUID = 6614257960044886319L;
 
+    @Override
+    public boolean canHandle(HttpServletRequest request) {
+
+        return isNativeSDKBasedFederationCall(request) || super.canHandle(request);
+    }
+
+    /**
+     * Check whether the authentication is based on API.
+     *
+     * @return true since API based authentication is supported.
+     */
+    @Override
+    public boolean isAPIBasedAuthenticationSupported() {
+
+        return true;
+    }
+
+    /**
+     * Get the i18n key defined to represent the authenticator name.
+     *
+     * @return the 118n key.
+     */
+    @Override
+    public String getI18nKey() {
+
+        return NaverCustomAuthenticatorConstants.AUTHENTICATOR_I18N_KEY;
+    }
+
+    @Override
+    protected void initiateAuthenticationRequest(HttpServletRequest request, HttpServletResponse response, AuthenticationContext context) throws AuthenticationFailedException {
+
+        if (logger.isDebugEnabled()) {
+            logger.debug("initiateAuthenticationRequest");
+        }
+
+        try {
+            Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
+            String clientId = (String)authenticatorProperties.get(CLIENT_ID);
+            String callbackUrl = (String)authenticatorProperties.get(CALLBACK_URL);
+            String authorizationEP = this.getAuthorizationServerEndpoint(authenticatorProperties);
+            String state = context.getContextIdentifier() + "," + NaverCustomAuthenticatorConstants.OAUTH2;
+            OAuthClientRequest authzRequest = OAuthClientRequest.authorizationLocation(authorizationEP).setClientId(clientId).setResponseType("code").setRedirectURI(callbackUrl).setState(state).buildQueryMessage();
+            if (logger.isDebugEnabled()) {
+                logger.debug("Authorization Request: " + authzRequest.getLocationUri());
+            }
+
+            context.setProperty(NaverCustomAuthenticatorConstants.OAUTH2_PARAM_STATE, state);
+            context.setProperty(AUTHENTICATOR_NAME + NaverCustomAuthenticatorConstants.REDIRECT_URL_SUFFIX,
+                    authzRequest.getLocationUri());
+            response.sendRedirect(authzRequest.getLocationUri());
+        } catch (IOException e) {
+            logger.error("Exception while sending to the login page.", e);
```

Comment:
why we need to log the error here.

Reply:
Fixed with https://github.com/wso2-extensions/identity-outbound-auth-naver/pull/13/commits/545814df82608605708a7e15c0ac5b8368dd8507